### PR TITLE
[RA-23651] Bound autosizing on vertical and horizontal

### PR DESCRIPTION
**Pull request by jsnyder4**
Column widths should autosize.
Row heights are not autosized. The height is determined by the new lines in the cell.

### CODE REVIEW

File: src/main/webapp/html/ntwobe.html [hunk 1]
```diff
@@ -47,6 +47,12 @@
                 </div>
             </div>
         </div>
+        <table class="handsontable" style="visibility: hidden">
+            <tr>
+                <td id="test-cell" class="cell"></td>
+                <td id="test-code" class="cell code"></td>
+            </tr>
+        </table>
```
Comment (jdereg) [inline src/main/webapp/html/ntwobe.html]: I see this is where the content is rendered and then measured.  Cool.

File: src/main/webapp/js/ntwobe.js [hunk 2]
```diff
@@ -779,10 +777,99 @@ var NCubeEditor2 = (function ($)
...
         setUpAxisColumnMap();
+        setUpColumnWidths();
+    };
+
+    var setUpColumnWidths = function()
+    {
+        var calcDomWidth = function (value, modifier, type)
+        {
+            var testElement = type == 'exp' ? testCode: testCell;
+            testElement[0].textContent = value;
```
Comment (jdereg) [inline src/main/webapp/js/ntwobe.js]: I would verify if .textContent is the right thing to use here, as opposed to .innerHtml.  Create a Groovy expression with something like 'println 'There is <html> in <b>this</b> string'.  If it displays correctly, leave it along.  If it has issues, it likely needs to use .innerHtml.

**Comment by jsnyder4** [comment]
I guess I should do my inline comments after the pull request. I thought I did, but maybe I did them on my branch so they don't show on the file diff here.

File: src/main/webapp/js/ntwobe.js [hunk 3]
```diff
@@ -779,10 +777,99 @@ var NCubeEditor2 = (function ($)
...
+                    var columnId, columnName;
+                    var axisName = axes[hotCol].name;
+                    var buttonWidth = calcDomWidth(axisName, 45, null);
+                    var oldWidth = findWidth(0, buttonWidth);
+
+                    var axisColumns = axisColumnMap[axisName];
+                    for (var axisCol = 0, axisColLength = axisColumns.length; axisCol < axisColLength; axisCol++)
+                    {
+                        columnId = axisColumns[axisCol];
+                        columnName = axes[hotCol].columns[columnId].value;
```
Comment (jdereg) [inline src/main/webapp/js/ntwobe.js]: columns can be URLs or Values.  Is this code handling that?

File: src/main/webapp/js/ntwobe.js [hunk 4]
```diff
@@ -779,10 +777,99 @@ var NCubeEditor2 = (function ($)
...
+                    var columnHeaderId = getColumnHeaderId(hotCol);
+                    _columnWidths.push(topWidths[columnHeaderId]);
+                }
+            }
+        };
+
+        _columnWidths = [];
+        var topWidths = {};
+        var testCell = $('#test-cell');
+        var testCode = $('#test-code');
```
Comment (jdereg) [inline src/main/webapp/js/ntwobe.js]: These variables are set here, but I can't tell what there scope is from the review.  My main pointi s...are they used?

File: src/main/webapp/js/ntwobe.js [hunk 5]
```diff
@@ -1754,6 +1837,7 @@ var NCubeEditor2 = (function ($)
         data.cells[_tableCellId] = {value:cellInfo.value};
         _cellId = null;
         destroyEditor();
+        reload();
```
Comment (tympollack) [inline src/main/webapp/js/ntwobe.js]: You should add reload to cell clear as well if you want it to recalculate.

**Pull request merged**